src/me/mrCookieSlime/ExoticGarden/Kitchen.java

```diff
@@ -0,0 +1,179 @@
+
+public class Kitchen {
+	
+	public static final RecipeType EG_KITCHEN = new RecipeType(new CustomItem(Material.CAULDRON_ITEM, "&eKitchen", 0, new String[] {"", "&a&oThis item should be made", "&a&oin the kitchen"}), "EG_KITCHEN");
+	
+	public static ItemStack KITCHEN = new CustomItem(Material.CAULDRON_ITEM, "&eKitchen", 0, new String[] {"", "&a&oYou can make a bunch of different yummies here"});
+	
+	@SuppressWarnings("deprecation")
+	public static void registerKitchen() {
+		//---------------------------------------------------------------------------------------------------------
+		new SlimefunMachine(Categories.MACHINES_1, KITCHEN, "KITCHEN",
+				new ItemStack[] {new CustomItem(Material.WOOD_STAIRS, "&oWooden stairs (upside down)", 1), new CustomItem(Material.WOOD_STAIRS, "&oWooden stairs (upside down)", 1), new ItemStack(Material.getMaterial(5)), new ItemStack(Material.WOOD_PLATE), new ItemStack(Material.IRON_TRAPDOOR), new ItemStack(Material.BOOKSHELF), new ItemStack(Material.FURNACE), new ItemStack(Material.DISPENSER), new ItemStack(Material.WORKBENCH)},
+				new ItemStack[0], Material.IRON_TRAPDOOR)
+				.register(true, new MultiBlockInteractionHandler() {
+
+					@Override
+					public boolean onInteract(Player p, MultiBlock mb, Block b) {
+
+						SlimefunMachine machine = (SlimefunMachine) SlimefunItem.getByID("KITCHEN");
+
+						if (mb.isMultiBlock(machine)) {
+							if (Slimefun.hasUnlocked(p, machine.getItem(), true)) {
+								Block raw_disp = b.getRelative(BlockFace.DOWN);
+								Dispenser disp = (Dispenser) raw_disp.getState();
+								final FurnaceInventory resinv;									
+								Furnace resf; 
+								
+								if (raw_disp.getRelative(BlockFace.EAST).getState().getBlock().getType().name() == "FURNACE") {
+									resf = (Furnace) raw_disp.getRelative(BlockFace.EAST).getState();
+									resinv = resf.getInventory();
+								} else if (raw_disp.getRelative(BlockFace.WEST).getState().getBlock().getType().name() == "FURNACE") {
+									resf = (Furnace) raw_disp.getRelative(BlockFace.WEST).getState();
+									resinv = resf.getInventory();
+								} else if (raw_disp.getRelative(BlockFace.NORTH).getState().getBlock().getType().name() == "FURNACE") {
+									resf = (Furnace) raw_disp.getRelative(BlockFace.NORTH).getState();
+									resinv = resf.getInventory();
+								} else  {
+									resf = (Furnace) raw_disp.getRelative(BlockFace.SOUTH).getState();
+									resinv = resf.getInventory();
+								}
+								
+								final Inventory inv = disp.getInventory();
+								List<ItemStack[]> inputs = RecipeType.getRecipeInputList(machine);
+
+								for (int i = 0; i < inputs.size(); i++) {
+									boolean craft = true;
+									for (int j = 0; j < inv.getContents().length; j++) {
+										if (!SlimefunManager.isItemSimiliar(inv.getContents()[j], inputs.get(i)[j], true)) {
+											craft = false;
+											break;
+										}
+									}
+									if (craft) {

```

Comment:
I tried to craft something in it (a jelly sandwich) and it says the recipe cannot be found.

Reply:
Thank you. It should work now :)